conanfile.py

```diff
@@ -161,12 +161,11 @@ def _build_msvc(self, args):
             else:
                 args += ["-openssl-linked"]
 
-            self.run("cd %s && %s && set" % (self.source_dir, vcvars))
-            self.run("cd %s && %s && configure %s"
-                     % (self.source_dir, vcvars, " ".join(args)))
-            self.run("cd %s && %s && %s %s"
-                     % (self.source_dir, vcvars, build_command, " ".join(build_args)))
-            self.run("cd %s && %s && %s install" % (self.source_dir, vcvars, build_command))
+            os.chdir(self.source_dir)
+            self.run("%s && set" % (vcvars))
```

Comment:
Instead of set command, I believe here we can do:
`with tools.environment_append()`
https://docs.conan.io/en/latest/reference/tools.html?highlight=tools#tools-environment-append

Reply:
After closer look, i see this is only printing the vars, not assigning them, and that environment_append is already used.

Reply:
I used this in mingw but in my memories I had to write this workaround on the msvc build. I will test it again, also there is others part that should need refactoring imo